packages/js/product-editor/src/components/block-editor/block-editor.tsx

```diff
@@ -142,8 +145,14 @@ export function BlockEditor( {
 
 	const { updateEditorSettings } = useDispatch( 'core/editor' );
 
+	const isEditorLoading =
+		! layoutTemplate ||
+		// variations don't have a product template
+		( postType !== 'product_variation' && ! productTemplate ) ||
```

Comment:
Isn't it safer to test if `postType === 'product'`?

Reply:
In theory an extension could add a custom post type for their product. We haven't really tested this out from our editor (and likely, it would break elsewhere in the classic experience I'd imagine), but right now, the only case where we are special handling the product template is when post type is `product_variation`, so I'd like to keep this check specific to that.

This is something we can and should revisit when we do more explorations into what the future or product types and traits, etc. looks like.